### DISCUSSION TITLE
message asynchronous response?

### DISCUSSION THREAD

**Discussion by ChuLiqiang** [resiprocate/resiprocate#341] [Q&A] [answered]
After receiving the SIP message, I need to reply with a 200 OK result only after another program returns the result. Is there a code example here that can asynchronously return the asynchronous result of the SIP message?

**Comment by sgodin** [comment, marked as answer]
Take a look at InviteSession::provideAnswerCommand and ServerInviteSession::acceptCommand API's.  These are safe to be called from any thread.

More info here:
https://github.com/resiprocate/resiprocate/wiki/DUM-Threading#user-content-Tips_on_Making_Thread_Safe_Calls_when_using_DUM
